File: WinForms/CS/AIMainDemo/CodeExamples.Modules/CodeExamples.cs
namespace DevExpress.AI.Demos {
    using System.Collections.Generic;
    using DevExpress.DXperience.Demos;
    using DevExpress.DXperience.Demos.CodeDemo;

    public partial class CodeExamples : CodeTutorialControlBase {
        static void CodeExample() {
            DevExpress.AIIntegration.WinForms.BehaviorInitializer.Initialize();//Force initialize Behavior in a case of starting code-example without running the demo
        }
        public CodeExamples() {
            InitializeComponent();
            #region Load External Assemblies
            string[] externalAssemblies = new string[] {
                typeof(DevExpress.Mvvm.ViewModelBase).Assembly.FullName,
                typeof(Microsoft.Extensions.DependencyInjection.ServiceCollection).Assembly.FullName,
                typeof(Microsoft.Extensions.DependencyInjection.ServiceCollectionContainerBuilderExtensions).Assembly.FullName,
                typeof(DevExpress.AIIntegration.WinForms.BehaviorInitializer).Assembly.FullName
            };
            foreach(string asmName in externalAssemblies)
                DevExpress.Data.Internal.SafeTypeResolver.GetOrLoadAssembly(asmName);
            #endregion Load External Assemblies
        }
        protected override string XtraItemGroupData {
            get {
                return $@"<PackageReference Include=""DevExpress.AIIntegration.WinForms"" Version=""//Version"" />";
            }
        }
        #region HighlightTokens
        protected override List<string> InitializeHighlightTokens() {
            var tokens = base.InitializeHighlightTokens();
            tokens.AddRange(DefaultTokens);
            return tokens;
        }
        readonly static string[] DefaultTokens = new string[] {
                "IChatClient",
                "AzureOpenAIClient",
                "ApiKeyCredential",
                "IAIExtensionsContainer",
                "AIExtensionsContainerDefault",
                "AIExtensionsContainerLocal",
                "AIExtensionsContainerDesktop",
                "AIExceptionHandler",
                "IAIExceptionHandler",
            };
        #endregion HighlightTokens
        protected override List<string> InitializeReferences() {
            List<string> _references = base.InitializeReferences();
            _references.AddRange(new string[] { });
            return _references;
        }
        protected override ExampleLanguage CurrentExampleLanguage {
            get { return DemoHelper.GetExampleLanguage(); }
        }
        protected override void DoHide() {
            base.DoHide();
            DestroyCurrentExample();
        }
    }
}

File: WinForms/VB/AIMainDemo/CodeExamples.Modules/CodeExamples.vb
Imports System.Collections.Generic
Imports DevExpress.DXperience.Demos
Imports DevExpress.DXperience.Demos.CodeDemo

Namespace DevExpress.AI.Demos

    Public Partial Class CodeExamples
        Inherits DevExpress.DXperience.Demos.CodeDemo.CodeTutorialControlBase

        Private Shared Sub CodeExample()
            DevExpress.AIIntegration.WinForms.BehaviorInitializer.Initialize() 'Force initialize Behavior in a case of starting code-example without running the demo
        End Sub

        Public Sub New()
            Me.InitializeComponent()
#Region "Load External Assemblies"
            Dim externalAssemblies As String() = New String() {GetType(DevExpress.Mvvm.ViewModelBase).Assembly.FullName, GetType(Microsoft.Extensions.DependencyInjection.ServiceCollection).Assembly.FullName, GetType(Microsoft.Extensions.DependencyInjection.ServiceCollectionContainerBuilderExtensions).Assembly.FullName, GetType(DevExpress.AIIntegration.WinForms.BehaviorInitializer).Assembly.FullName}
            For Each asmName As String In externalAssemblies
                DevExpress.Data.Internal.SafeTypeResolver.GetOrLoadAssembly(asmName)
            Next
#End Region  ' Load External Assemblies
        End Sub

        Protected Overrides ReadOnly Property XtraItemGroupData As String
            Get
                Return $"<PackageReference Include=""DevExpress.AIIntegration.WinForms"" Version=""//Version"" />"
            End Get
        End Property

#Region "HighlightTokens"
        Protected Overrides Function InitializeHighlightTokens() As List(Of String)
            Dim tokens = MyBase.InitializeHighlightTokens()
            tokens.AddRange(DevExpress.AI.Demos.CodeExamples.DefaultTokens)
            Return tokens
        End Function

        Private ReadOnly Shared DefaultTokens As String() = New String() {"IChatClient", "AzureOpenAIClient", "ApiKeyCredential", "IAIExtensionsContainer", "AIExtensionsContainerDefault", "AIExtensionsContainerLocal", "AIExtensionsContainerDesktop", "AIExceptionHandler", "IAIExceptionHandler"}

#End Region  ' HighlightTokens
        Protected Overrides Function InitializeReferences() As List(Of String)
            Dim _references As System.Collections.Generic.List(Of String) = MyBase.InitializeReferences()
            _references.AddRange(New String() {})
            Return _references
        End Function

        Protected Overrides ReadOnly Property CurrentExampleLanguage As ExampleLanguage
            Get
                Return DevExpress.DXperience.Demos.DemoHelper.GetExampleLanguage()
            End Get
        End Property

        Protected Overrides Sub DoHide()
            MyBase.DoHide()
            Me.DestroyCurrentExample()
        End Sub
    End Class
End Namespace
